lib/ui/compositing/scene_host.cc

```diff
@@ -10,33 +10,49 @@
 #include <third_party/tonic/dart_binding_macros.h>
 #include <third_party/tonic/logging/dart_invoke.h>
 
+#include "dart/runtime/include/dart_api.h"
 #include "flutter/flow/view_holder.h"
 #include "flutter/fml/thread_local.h"
 #include "flutter/lib/ui/ui_dart_state.h"
 
 namespace {
 
-using SceneHostBindings = std::unordered_map<zx_koid_t, flutter::SceneHost*>;
+struct SceneHostBindingKey {
+  Dart_Isolate associated_isolate;
```

Comment:
Call `Dart_IsolateServiceId` and use that to represent the isolate.  A raw isolate handle can become invalid at any time.

Reply:
good call, thanks @jason-simmons